tasks/eva_3033/detect_project_status.py

```diff
@@ -0,0 +1,228 @@
+import argparse
+import glob
+import gzip
+import os.path
+
+from cached_property import cached_property
+
+from ebi_eva_common_pyutils.logger import logging_config
+from ebi_eva_common_pyutils.metadata_utils import get_metadata_connection_handle
+from ebi_eva_common_pyutils.mongo_utils import get_mongo_connection_handle
+from ebi_eva_common_pyutils.pg_utils import get_all_results_for_query, execute_query
+
+
+logger = logging_config.get_logger(__name__)
+logging_config.add_stderr_handler()
+
+
+def detect_all_public_project(maven_config, maven_profile):
+    with get_metadata_connection_handle(maven_profile, maven_config) as pg_conn:
+        query = f"select project_accession from evapro.project where ena_status=4"
+        public_projects = [project for project, in get_all_results_for_query(pg_conn, query)]
+
+    return public_projects
+
+
+def process_projects(projects, maven_config, maven_profile, noah_project_dir, codon_project_dir):
+    print('\t'.join(["Project", "Analysis", "Taxonomy", "Submitted assembly", "Remapped assembly",
+                     "Accessioning status", "Accessioning ssid found", "Remapping status", "Remapping ssid found",
+                     "Clustering status", "Clustering ssid found"]))
+    for project in projects:
+        detector = ProjectStatusDetector(project, maven_config, maven_profile, noah_project_dir, codon_project_dir)
+        detector.detect_project_status()
+
+
+class ProjectStatusDetector:
+
+    def __init__(self, project, maven_config, maven_profile, noah_project_dir, codon_project_dir):
+        self.project = project
+        self.maven_config = maven_config
+        self.maven_profile = maven_profile
+        self.noah_project_dir = noah_project_dir
+        self.codon_project_dir = codon_project_dir
+
+    @cached_property
+    def metadata_conn(self):
+        return get_metadata_connection_handle(self.maven_profile, self.maven_config)
+
+    @cached_property
+    def mongo_conn(self):
+        return get_mongo_connection_handle(self.maven_profile, self.maven_config)
```

Comment:
Suggest setting majority read/write concern here if possible..

Reply:
I'm starting to sense this is a common comment by Sundar on our Python PRs, so I made [this PR](https://github.com/EBIvariation/eva-common-pyutils/pull/31) on the common library to set some sensible defaults... we usually do this on a per-operation level anyway but maybe this will reduce the possibility of mistakes...